tools/hardhat-example/contracts/SimpleVault.sol

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
```

Comment:
Q: what's the goal with the `hardhat-example` update?
Tool examples in this repo are supposed to show simple coverage that the tool in this case hardhat works with the relay. It doesn't need to have additional contract testing.
If there's a gap in contract coverage that should go in the smart-contracts repo.
In this repo the additional tests should be around the relay API which would be the in the main `test` folder and not under a single tool.

Either ways this change would have to be pulled out into it's own PR as we try to keep PRs focused  for easier review and tracking.

Reply:
`hardhat-example` changes are now on [this PR](https://github.com/mshakeg/hedera-json-rpc-relay/pull/27)

Reply:
> It's to simply demo how the system contract mocks can be used in a hardhat project:
> 
> 
> 
> 1. install [hashgraph/hedera-smart-contracts](https://github.com/hashgraph/hedera-smart-contracts) as a dependency
> 
> 2. In your contracts use the various Hedera system contract abstract contracts/interfaces as you normally would as demonstrated in the `SimpleVault` example contract
> 
> 3. In your JS/TS test scripts deploy the requisite Hedera system contract mocks to their respective pre-defined addresses using `hardhat_setCode`
> 
> 4. You can now FULLY test your Hedera specific smart contracts against a hardhat local node, whereas before you had to connect to a Hedera local node/network, which is way slower and doesn't have the many debugging methods of the hardhat local node.
> 
> 
> 
> ok, I'll create a separate PR titled "demo system contract mocks in hardhat example"

Okay. Great.
Initially there were concerns that system contracts would not work since a Hardhat local node does not have the Hedera Java logic that matches the contracts.
Are you saying this approach isn't affected by that.
If it's just a regular non system contract then hardhat and other tools would work fine with a Hedera relay. However, I'm curious about the above.

Thanks for breaking it out into a separate PR, will take a look.

Reply:
> Initially there were concerns that system contracts would not work since a Hardhat local node does not have the Hedera Java logic that matches the contracts. Are you saying this approach isn't affected by that.

That's still an issue, but that's not what the addition to the `hardhat-example` does. Rather it's a demo of how developers could avoid having to use a Hedera local node(or network) in their hardhat projects by instead deploying the newly added **system contracts mocks** defined in `hedera-smart-contracts` to their local hardhat node and effectively mock the environment of a Hedera node.

Hope that clarifies it.

Reply:
Sounds good, will watch out for PR submission here later if you want to bring that along to enhance the hardhat project example.
Thanks

Reply:
will do, I was considering implementing the `hardhat-example` in TS and JS is a pain to develop in. I was also surprised that the `hedera-smart-contract` hardhat project doesn't use TS.